### ISSUE TITLE
Updating from 2016-10-28: black/blank screen (no hint to key Control-Alt-F2; no caution against powering off the computer)

### ISSUE THREAD

**Issue by grahamperrin** [trueos/trueos-core#171]
Problem observed with an HP EliteBook 850 G2 –

- Intel Core i7-5500U with Intel HD Graphics 5500,
- AMD Radeon R7 M260X (1 GB DDR5 dedicated, dual graphics) – discrete

```
$ date ; uname -a
Sat 26 Nov 2016 16:19:11 GMT
FreeBSD momh167-gjp4-hpelitebook850g2-trueos.university.brighton.ac.uk 12.0-CURRENT FreeBSD 12.0-CURRENT #14 1717ae3(drm-next-4.7): Fri Oct 28 13:49:49 UTC 2016     root@gauntlet:/usr/obj/usr/src/sys/GENERIC  amd64
$ grep kms /var/log/Xorg.0.log
[   178.162] (II) modesetting: Driver for Modesetting Kernel Drivers: kms
$ xrandr
Screen 0: minimum 320 x 200, current 1366 x 768, maximum 8192 x 8192
DP-1 disconnected (normal left inverted right x axis y axis)
HDMI-1 disconnected (normal left inverted right x axis y axis)
DP-2 connected 1366x768+0+0 (normal left inverted right x axis y axis) 410mm x 230mm
   1366x768      59.79*+
   1280x800      59.81  
   1024x768      75.03    60.00  
   800x600       75.00    60.32  
   640x480       75.00    59.94  
   720x400       70.08  
HDMI-2 disconnected (normal left inverted right x axis y axis)
$ 
```

(Before loading kernel modules, the OS drives the integral display but no external display. After loading modules, external displays alone are driven. More specifically: greyness at the notebook before and during both stages of the update routine, blackness at the external display when stage two begins.)

## Workaround

_Without observing a hint_ to key Control-Alt-F2, perform that key combination. 

----

https://forums.pcbsd.org/thread-21075-post-115489.html#pid115489 seems to exemplify an incident with [NVIDIA GeForce GTS 450](http://www.geforce.co.uk/hardware/desktop-gpus/geforce-gts-450) where Control-Alt-F2 does not reveal progress; blackness persisted. 

----

Not reproducible with: 

- MacBookPro11,2

**Comment by pkgdemon** [comment]
Sounds like we should try to broadcast the "press f2" message across all screens?  Would that solve it?

**Comment by grahamperrin** [comment]
I wonder. 

If it helps: I'm almost certain that with the HP EliteBook 850 G2, the behaviour is regressive; there _was_ the customary alert during stage two of an update from … I'm not sure what, to (2016-10-28) build 14 of the OS.

**Comment by pkgdemon** [comment]
I have a system with a 4k monitor that acts really strange when trying to switch to a VT.  It goes dark, and I end up having to reset. I am not sure if it the same issue but I will take a closer look to see if it can reproduce.

**Comment by grahamperrin** [comment]
> blackness at the external display when stage two begins.

More accurately, AFAIR: 

- a white movable pointer arrow
- a white block cursor at top left (I can't recall whether it blinked)

– not complete blackness. 

----

With one of my more recent experiments with the 850 G2, maybe whilst updating with an already UNSTABLE version of TrueOS and two displays at DisplayPort sockets of the HP dock, if I recall correctly the transition was: 

- from PCDM (after log out from KDE)
- to the detailed view of stage two

– neither blackness, nor the hint to key Control-Alt-F2 for details.

**Comment by pkgdemon** [comment]
Does this issue involve KDE only?  Is it still a with Lumina, and more recent updates?

**Comment by grahamperrin** [comment]
No recent incidents, I can not recall when I last attempted an update from 2016-10-28. I might do so in a few hours (this evening). 

As far as I recall: whenever it occurred, it was after logging out from KDE (because the GUI in KDE is without options to shut down or restart). So I would log out, then at PCDM either (a) key Escape to shut down; or (b) use the power menu to restart.

**Comment by pkgdemon** [comment]
Ah.  So maybe it is triggered by using PCDM to restart, or shutdown.  I will give that a try the next update.

**Comment by grahamperrin** [comment]
>> … I can not recall when I last attempted an update from 2016-10-28. I might do so in a few hours …

Done. 

> … maybe it is triggered by using PCDM to restart, or shutdown. …

Nope, I used the GUI of Lumina to request a restart, the problem recurred. 

Cases such as this may be rare (I noticed just one other possible incident affecting another user) – and not reproducible after the update – so rather than work towards a fix at this time, I suggest simply making users aware of the _possibility_ of encountering this bug. Maybe a 'Countdown to stable' topic (or words to that effect) in the [announcements](https://discourse.trueos.org/c/announcements) area, a couple of days before the push, outlining known issues.
